### DISCUSSION TITLE
What are some ways to reduce the size of `"vercel_app/wsgi.py"`?

### DISCUSSION THREAD

**Discussion by hirehamir** [vercel/community#2070] [Help] [answered]
### Problem Description

Things were working fine until I decided to add `pydub` as a dependency, at which point running the `vercel` command failed with this error.

```
Error: The Serverless Function "vercel_app/wsgi.py" is 70.22mb which exceeds the maximum size limit of 50mb.
```

I need `pydub` since it lets me convert files from `.m4a` to `.mp3`, which I need to do for iPad OS and iOS Safari to display audio durations correctly.

`vercel_app/wsgi.py` is a small file in my Django project and I don't see where Vercel gets `70.22mb` from.

I think verbose command line output here instructing users how to make their serverless functions smaller would definitely be helpful.

Deleting dependencies from `requirements.txt` via trial-and-error will at least get the `vercel` command to complete without error, but then visiting the development link results in the serverless function crashing due to an `ImportModuleError`.

Contents of `vercel_app/wsgi.py`:

```py
"""
WSGI config for vercel_app project.

It exposes the WSGI callable as a module-level variable named ``app``.

For more information on this file, see
https://docs.djangoproject.com/en/4.1/howto/deployment/wsgi/
"""

import os

from django.core.wsgi import get_wsgi_application

os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'vercel_app.settings')

app = get_wsgi_application()
```

**Comment by afjol77** [comment]
Facing the same problem. Tried in many ways, but not getting any solution. Did you fix it? Please let me know

**Reply by hirehamir** [reply]
I haven't found any fixes. Removing dependencies from `requirements.txt` might help?

**Reply by amyegan** [reply, marked as answer]
Dependencies and imported files are common culprits. For specific files, you can add [functions](https://vercel.com/docs/concepts/projects/project-configuration#functions) configuration with `includeFiles` and `excludeFiles` to help adjust the function size - https://github.com/vercel/community/discussions/103#discussioncomment-1052162
